kube-runtime/src/watcher.rs

```diff
@@ -181,10 +181,37 @@ async fn step<K: Meta + Clone + DeserializeOwned + Send + 'static>(
     }
 }
 
-/// Watches a Kubernetes Resource for changes
+/// Watches a Kubernetes Resource for changes continuously
 ///
-/// Errors are propagated to the client as `Err`. Tries to recover (by reconnecting and resyncing as required)
-/// if polled again after an error.
+/// Creates an indefinite read stream through continual [`Api::watch`] calls, and keepint track
+/// of [returned resource versions](https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes).
+/// It tries to recover (by reconnecting and resyncing as required) if polled again after an error.
+///
+/// This is intended to provide a safe input interface for a state store like a [`reflector`],
+/// so if used directly, it's recommended to use it with with [`try_flatten_applied`]:
+///
+/// ```no_run
+/// use kube::{api::{Api, ListParams, Meta}, Client};
+/// use kube_runtime::{utils::try_flatten_applied, watcher};
+/// use k8s_openapi::api::core::v1::Pod;
+/// use futures::{StreamExt, TryStreamExt};
+/// #[tokio::main]
+/// async fn main() -> Result<(), kube_runtime::watcher::Error> {
+///     let client = Client::try_default().await.unwrap();
+///     let pods: Api<Pod> = Api::namespaced(client, "apps");
+///     let watcher = watcher(pods, ListParams::default());
+///     try_flatten_applied(watcher)
+///         .try_for_each(|p| async move {
+///          println!("Applied: {}", Meta::name(&p));
+///             Ok(())
+///         })
+///         .await?;
+///    Ok(())
+/// }
+/// ```
+/// [`try_flatten_applied`]: super::utils::try_flatten_applied
+/// [`reflector`]: super::reflector::reflector
+/// [`Api::watch`]: https://docs.rs/kube/*/kube/struct.Api.html#method.watch
```

Comment:
I thought rustdoc was supposed to resolve these links automagically these days

Reply:
I wasn't able to get a cross-crate link to work :/

Reply:
Seems to work for tokio-util (https://docs.rs/tokio-util/0.6.0/src/tokio_util/context.rs.html#22).. which direction did you test in? Makes sense that it would be tricker for links from `kube` -> `kube-runtime` than the other way around.

Reply:
They have the `Runtime` in scope 7 lines up. If we ever re-export `kube_runtime` from `kube` under a `runtime` feature (which I think we should aim to do), then within crate links will work from `kube` -> `kube_runtime`.

The other way around has the same problem for the time being.